zerver/lib/actions.py

```diff
@@ -4462,20 +4462,28 @@ def get_email_gateway_message_string_from_address(address: str) -> Optional[str]
 
     return msg_string
 
-def decode_email_address(email: str) -> Optional[Tuple[str, str]]:
-    # Perform the reverse of encode_email_address. Returns a tuple of (streamname, email_token)
+def decode_email_address(email: str) -> Optional[Tuple[str, str, bool]]:
+    # Perform the reverse of encode_email_address. Returns a tuple of
+    # (streamname, email_token, show_sender)
     msg_string = get_email_gateway_message_string_from_address(email)
-
     if msg_string is None:
         return None
-    elif '.' in msg_string:
+
+    if msg_string.endswith('show-sender'):
```

Comment:
This should be `+show-sender`, right?

Reply:
I think we want to support both using pluses and dots in msg_string, so I adjusted for this fact and changed this line to
`` if msg_string.endswith(('+show-sender', '.show-sender')): ``